server/rest/src/main/java/org/infinispan/rest/resources/ServerResource.java

```diff
@@ -332,26 +344,82 @@ private CompletionStage<RestResponse> threads(RestRequest restRequest) {
 
    private CompletionStage<RestResponse> report(RestRequest restRequest) {
       ServerManagement server = invocationHelper.getServer();
-      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
       return Security.doAs(restRequest.getSubject(), (PrivilegedAction<CompletionStage<RestResponse>>) () ->
             server.getServerReport().handle((path, t) -> {
                if (t != null) {
```

Comment:
I know it wasn't like this before, but shouldn't we at least log the error ? Or is it logged somewhere else?

Reply:
Looking closer, I think the approach here is to throw so it is caught and logged at 
https://github.com/infinispan/infinispan/blob/9bdcf91c2cd99353e483a63905f191bfc290bfdc/server/rest/src/main/java/org/infinispan/rest/BaseHttpRequestHandler.java#L43-L44

Reply:
That code will never be hit. This is just invoking `getServerReport` https://github.com/infinispan/infinispan/blob/main/server/runtime/src/main/java/org/infinispan/server/Server.java#L671.

Since this is chained using the `handle` method the exception is lost and translated into a simple server error only. I don't see how it is handled at all, it could be a server report unavailable, io exception, timeout exception or any other runtime exception that is swallowed.

Reply:
Do you mean we should return distinct HTTP codes? In the current approach, we return the exception message but that's all.

Reply:
The code looks good now. My original point was the way it was working before the latest changes that it ignored the t variable completely.